src/Compilers/Core/Portable/Syntax/SyntaxTriviaList.cs

```diff
@@ -236,33 +237,71 @@ public SyntaxTriviaList Insert(int index, SyntaxTrivia trivia)
             return InsertRange(index, new[] { trivia });
         }
 
+        private static readonly ObjectPool<SyntaxTriviaListBuilder> s_builderPool =
+            new ObjectPool<SyntaxTriviaListBuilder>(() => SyntaxTriviaListBuilder.Create());
+
+        private static SyntaxTriviaListBuilder GetBuilder()
+            => s_builderPool.Allocate();
+
+        private static void ClearAndFreeBuilder(SyntaxTriviaListBuilder builder)
+        {
+            // It's possible someone might create a list with a huge amount of trivia
+            // in it.  We don't want to hold onto such items forever.  So only cache
+            // reasonably sized lists.
+            const int MaxBuilderCount = 16;
```

Comment:
So, iirc around 75% of all trivia lists were <8.  Basically, you get a fair amount of cases (like comment sequences) that bump things above 8.  For example, a simple chain of:

```
    // foo
    // bar
    // baz
```

Is already at 9 trivia since you have ```<whitespace><comment><end of line>``` for each line.